src/parser/postgresparser.cpp

```diff
@@ -288,36 +288,23 @@ parser::GroupByDescription* PostgresParser::GroupByTransform(List* group,
   result->columns = new std::vector<expression::AbstractExpression*>();
   for (auto cell = group->head; cell != nullptr; cell = cell->next) {
     Node* temp = reinterpret_cast<Node*>(cell->data.ptr_value);
-    switch (temp->type) {
-      case T_ColumnRef:
```

Comment:
You are right. GroupByTransform and OrderByTransform should not call the ExprTransform directly but in other cases it should be fine. Note that group by and order by can have constant expressions to directly indicate the column positions, which we may overlook in the new optimizer.